utils/reference_checker.py

```diff
@@ -155,6 +168,120 @@ def get_citation(paper):
     citation = f"{names}. {corrected_title} {journal} {volume} ({pub_date[0:4]}):{pages}"
     return citation
 
+## 12/18/2023 IL updated the following main function
+def check_title_matching(paper, paragraph, config, n=10, verbose=False):
+    ## load config for openai query
+    context = config['CONTEXT']
+    gpt_model = config['GPT_MODEL']
+    temperature = config['TEMP']
+    max_tokens = config['MAX_TOKENS']
+    rate_per_token = config['RATE_PER_TOKEN']
+    LOG_FILE = config['LOG_NAME']+'log.json'
+    DOLLAR_LIMIT = config['DOLLAR_LIMIT']
+    
+    try:
+        title = paper['MedlineCitation']['Article']['ArticleTitle']
+    except (KeyError, IndexError) as e:
+        if verbose:
+            print("Error in getting abstract from paper.")
+            print("Error detail: ", e)
+        return False
+    message = f"""
+I have a paragraph
+Paragraph:
+{paragraph}
+
+and an scientific paper title.
+Abstract:
+{title}
+
+Does this title support paragraph? Please tell me yes or no
+"""    
+    try:
+        result = openai_chat(context, message, gpt_model, temperature, max_tokens, rate_per_token, LOG_FILE, DOLLAR_LIMIT)
+    except Exception as e:
+        print("Error in openai_chat")
+        print("Error detail: ", e)
+        result = None
+        return False
+    if result[:3].lower()=='yes':
+        return True
+    else:
+        return False
+
+## 12/18/2023 IL updated the following main function
+def check_abstract_match(paper, paragraph, config, verbose=False):
+    ## load config for openai query
+    context = config['CONTEXT']
+    gpt_model = config['GPT_MODEL']
+    temperature = config['TEMP']
+    max_tokens = config['MAX_TOKENS']
+    rate_per_token = config['RATE_PER_TOKEN']
+    LOG_FILE = config['LOG_NAME']+'log.json'
+    DOLLAR_LIMIT = config['DOLLAR_LIMIT']
+
+    try:
+        abstract = "\n".join(paper['MedlineCitation']['Article']['Abstract']['AbstractText'])
+    except (KeyError, IndexError) as e:
+        if verbose:
+            print("Error in getting abstract from paper.")
+            print("Error detail: ", e)
+        return False
+    message = f"""
+I have a paragraph
+Paragraph:
+{paragraph}
+
+and an abstract.
+Abstract:
+{abstract}
+
+Does this abstract support paragraph? Please tell me yes or no
+        """
+        
+    try:
+        result = openai_chat(context, message, gpt_model, temperature, max_tokens, rate_per_token, LOG_FILE, DOLLAR_LIMIT)
```

Comment:
I have updated the openai_chat function lately, do you mind updating this line accordingly and give a test run?